### ISSUE TITLE
Two graph  matching 

### ISSUE THREAD

**Issue by xiachenrui** [Thinklab-SJTU/pygmtools#9]
Hi, thanks for your great works! I am interested in using it. But when I try to match 2 graphs use your [demo](https://pygmtools.readthedocs.io/en/latest/api/_autosummary/pygmtools.multi_graph_solvers.gamgm.html#pygmtools.multi_graph_solvers.gamgm)  , it seems not work:

```python
import torch
import pygmtools as pygm
import itertools
import time

from zmq import device
pygm.BACKEND = 'pytorch'
_ = torch.manual_seed(1)

# Generate 2 isomorphic graphs
graph_num = 2
node_num = 100
device = 'cpu'

# As: adjacency matrix; X_gt: groud truth mapping matrix; Fs: features matrix
As, X_gt, Fs = pygm.utils.generate_isomorphic_graphs(node_num=node_num, graph_num=graph_num, node_feat_dim=20)

# Compute node-wise similarity by inner-product and Sinkhorn
W = torch.matmul(Fs.unsqueeze(1), Fs.transpose(1, 2).unsqueeze(0))
W = pygm.sinkhorn(W.reshape(graph_num ** 2, node_num, node_num)).reshape(graph_num, graph_num, node_num, node_num)

# Solve the multi-matching problem
As = As.to(device)
W = W.to(device)
X = pygm.gamgm(As, W)
matched = 0
for i, j in itertools.product(range(graph_num), repeat=2):
    matched += (X[i,j] * X_gt[i,j]).sum()
matched / X_gt.sum()
------------------------------------------------------------------------
>>> tensor(0.)

```

But if I only change the  `graph_num` to 10 it seems works and the output is 
```python
tensor(0.1108)
```

By the way , I would like to ask, what is currently the best way to unsupervised matching two graphs. Graphs come from high-resolution medical images. Each graph contains ~30,000 nodes, and each node is evenly arranged on the image and contains 50 features. I guess this is not the same as your paper's solution of extracting a small number of keypoints on an image

Thanks again ！

Xia Chenrui

**Comment by rogerwwww** [comment]
Hi Chenrui, 

This behavior is expected because this algorithm is sensitive to hyperparameters (converging fast while still preserving the accuracy). Since you have modified the number of nodes and graphs, probably you need a new set of hyperparameters. Setting ``verbose=True`` will print useful information that helps you tune the parameters.

For matching larger graphs, perhaps you can start with the two-graph matching methods in classic_solvers.py

**Comment by xiachenrui** [comment]
Thanks a lot! I will try it
